lib/ansible/modules/cloud/vmware/vcenter_folder.py

```diff
@@ -163,7 +163,7 @@ def ensure(self):
         folder_type = self.module.params.get('folder_type')
         folder_name = self.module.params.get('folder_name')
         parent_folder = self.module.params.get('parent_folder', None)
-        results = dict(changed=False, result=dict())
+        results = {'changed': False, 'msg': ''}
```

Comment:
I would suggestion to keep `result=dict()` as it is for backward compatibility. Add new keys `path` and `msg` as new keys in dict.

Reply:
Do we really need to keep the backward compat here? result was just a status string.

Reply:
Yes, since this module may be used by a lot of people. And this change may fail playbooks for them.

Reply:
Is the patch ok for you now? From my view point, if we want to expose a new key, we have to change result type from str to dict, and so we break the compat.